aries_vcx/src/protocols/proof_presentation/prover/state_machine.rs

```diff
@@ -131,18 +259,16 @@ impl ProverSM {
             source_id,
             state,
             thread_id,
-        } = self;
+        } = self.clone();
```

Comment:
This clone is something you've eliminated from issuer/holder. Get we get by without it in prover/verifier as well?